server/http.go

```diff
@@ -349,10 +571,69 @@ func (p *Plugin) postMeeting(creator *model.User, meetingID int, channelID strin
 	if appErr = p.storeMeetingPostID(meetingID, createdPost.Id); appErr != nil {
 		p.API.LogDebug("failed to store post id", "err", appErr)
 	}
-
 	return nil
 }
 
+func (p *Plugin) askUserForMeetingPreference(userID, channelID string) {
+	apiEndPoint := fmt.Sprintf(APIToAskForPMI, manifest.ID)
+
+	slackAttachment := model.SlackAttachment{
+		Pretext: AskForPMIMeeting,
+		Actions: []*model.PostAction{
+			{
+				Id:    "WithPMI",
+				Name:  UsePersonalMeetingID,
+				Type:  model.PostActionTypeButton,
+				Style: "default",
+				Integration: &model.PostActionIntegration{
+					URL: apiEndPoint,
+					Context: map[string]interface{}{
+						Action:    UsePersonalMeetingID,
+						UserID:    userID,
+						ChannelID: channelID,
+					},
+				},
+			},
+			{
+				Id:    "WithoutPMI",
+				Name:  UseAUniqueMeetingID,
+				Type:  model.PostActionTypeButton,
+				Style: "default",
+				Integration: &model.PostActionIntegration{
+					URL: apiEndPoint,
+					Context: map[string]interface{}{
+						Action:    UseAUniqueMeetingID,
+						UserID:    userID,
+						ChannelID: channelID,
+					},
+				},
+			},
+		},
+	}
+
+	post := &model.Post{
+		ChannelId: channelID,
+		UserId:    p.botUserID,
+	}
+	model.ParseSlackAttachment(post, []*model.SlackAttachment{&slackAttachment})
+	p.API.SendEphemeralPost(userID, post)
+}
+
+func (p *Plugin) getPMISettingData(userID string) (string, error) {
+	preferences, reqErr := p.API.GetPreferencesForUser(userID)
+	if reqErr != nil {
+		return "", errors.New(SettingDataError)
+	}
+
+	for _, preference := range preferences {
+		if preference.UserId != userID || preference.Category != zoomPreferenceCategory || preference.Name != zoomPMISettingName {
+			continue
+		}
+		return preference.Value, nil
```

Comment:
I think this reads a little better

```suggestion
		if preference.UserId == userID && preference.Category == zoomPreferenceCategory && preference.Name == zoomPMISettingName {
		    return preference.Value, nil
		}
```